File: sdks/flutter/lib/deepkick.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

class DeepKickClient {
  final String baseUrl;

  DeepKickClient({this.baseUrl = 'http://localhost:8080/v1'});

  Future<Map<String, dynamic>> getPrediction(String matchId) async {
    final response = await http.get(Uri.parse('$baseUrl/match/$matchId/predict'));
    if (response.statusCode == 200) {
      return json.decode(response.body);
    } else {
      throw Exception('Failed to load prediction');
    }
  }

  Future<Map<String, dynamic>> getPlayerForm(String playerId) async {
    final response = await http.get(Uri.parse('$baseUrl/player/$playerId/form'));
    if (response.statusCode == 200) {
      return json.decode(response.body);
    } else {
      throw Exception('Failed to load player form');
    }
  }
}

File: sdks/python/deepkick.py
import requests

class DeepKickClient:
    """DeepKick Python SDK Client"""
    def __init__(self, base_url="http://localhost:8080/v1"):
        self.base_url = base_url

    def get_prediction(self, match_id: str) -> dict:
        url = f"{self.base_url}/match/{match_id}/predict"
        response = requests.get(url)
        response.raise_for_status()
        return response.json()

    def get_player_form(self, player_id: str) -> dict:
        url = f"{self.base_url}/player/{player_id}/form"
        response = requests.get(url)
        response.raise_for_status()
        return response.json()
